CRM/Contribute/BAO/Contribution.php

```diff
@@ -374,7 +374,8 @@ public static function calculateMissingAmountParams(&$params, $contributionID) {
         $params['fee_amount'] = 0;
       }
     }
-    if (!isset($params['net_amount'])) {
+    // recalculate net amount if it is not set or if tax amount is set
+    if (!isset($params['net_amount']) || isset($params['tax_amount'])) {
```

Comment:
There was discussion on how net amount relates to tax_amount elsewhere , I feel both @KarinG and @JoeMurray argued it was net of fees, not of taxes